server/src/test/java/org/elasticsearch/index/mapper/RootObjectMapperTests.java

```diff
@@ -669,9 +683,9 @@ public void testRuntimeSectionMissingType() throws IOException {
     }
 
     public void testRuntimeSectionWrongFormat() throws IOException {
-        XContentBuilder mapping = runtimeMapping(builder -> builder.field("field", "value"));
+        XContentBuilder mapping = runtimeMapping(builder -> builder.field("field", 123));
         MapperParsingException e = expectThrows(MapperParsingException.class, () -> createMapperService(mapping));
-        assertEquals("Failed to parse mapping: Expected map for runtime field [field] definition but got a java.lang.String",
+        assertEquals("Failed to parse mapping: Expected map for runtime field [field] definition but got a java.lang.Integer",
```

Comment:
Is this related to the rest of the PR?

Reply:
good question: a little, or maybe not. We used to error when null was provided (same error as when something that is not a map is provided), but I figured the error message contained the type of the key (always string) rather than that of the value :) We even had a test for it which was working because the value was also a string, so I changed it to integer just to make sure.